google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -166,33 +171,23 @@ public JComponent createComponent() {
   @Override
   public boolean isModified() {
     return !appYaml.getText().equals(facetConfiguration.getAppYamlPath())
-        || !dockerfile.getText().equals(facetConfiguration.getDockerfilePath());
+        || !dockerDirectory.getText().equals(facetConfiguration.getDockerDirectory());
   }
 
   @Override
   public void reset() {
     appYaml.setText(facetConfiguration.getAppYamlPath());
-    dockerfile.setText(facetConfiguration.getDockerfilePath());
+    dockerDirectory.setText(facetConfiguration.getDockerDirectory());
 
     toggleDockerfileSection();
   }
 
   @Override
   public void apply() throws ConfigurationException {
     showWarnings();
-    if (!isValidConfigurationFile(appYaml.getText())) {
-      throw new ConfigurationException(
-          GctBundle.getString("appengine.deployment.error.staging.yaml"));
-    }
-
-    try {
-      if (isRuntimeCustom() && !isValidConfigurationFile(dockerfile.getText())) {
-        throw new ConfigurationException(
-            GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
-      }
-    } catch (MalformedYamlFileException myf) {
-      throw new ConfigurationException(
-          GctBundle.getString("appengine.appyaml.malformed"));
+    Result result = validateConfiguration();
```

Comment:
why call validateConfiguration twice? showWarnings already calls it. I know that showWarnings is called from other places than apply(). You could do something like return the validationResult from showWarnings (and rename it to validateAndShowWarnings, or something).